tests/unit/components/reception/SharedTestsForImportAndScanIn.js

```diff
@@ -0,0 +1,166 @@
+// Shared tests for the functionality of importing and scanning labware into LabWhere,
+// available on the Reception page
+//
+import * as Reception from '@/services/traction/Reception.js'
+import { expect, it } from 'vitest'
+import { store } from '@support/testHelper.js'
+
+export const sharedTestsForImportAndScanIn = (
+  wrapper,
+  scanBarcodesInLabwhereLocation,
+  mockShowAlert,
+) => {
+  describe('Unsuccessful Import ', () => {
+    vi.mock('@/services/labwhere/client.js')
+    vi.spyOn(console, 'error').mockImplementation(() => {})
+
+    describe('Fails to find labware', () => {
+      beforeEach(async () => {
+        wrapper.vm.reception.fetchFunction.mockRejectedValue('Failed fetch')
+        await wrapper.vm.fetchLabware()
+        wrapper.vm.reception.fetchFunction = vi.fn()
+        vi.spyOn(Reception, 'createReceptionResource').mockImplementation(() => {})
+      })
+      it('shows error alert message', () => {
+        expect(mockShowAlert).toHaveBeenCalledWith('Failed fetch', 'danger')
+      })
+
+      it('does not call create resource functionality', async () => {
+        expect(Reception.createReceptionResource).not.toHaveBeenCalled()
+      })
+
+      it('does call scan in LabWhere functionality', () => {
+        expect(scanBarcodesInLabwhereLocation).not.toHaveBeenCalled()
+      })
+    })
+
+    describe('Fails to import labware', () => {
+      store.state.traction.messages = []
+
+      it('calls create resource functionality', async () => {
+        await wrapper.vm.importLabware()
+        expect(Reception.createReceptionResource).toHaveBeenCalled()
+      })
+
+      it('shows error alert message', async () => {
+        const message = 'The princess is in another castle'
+        vi.spyOn(Reception, 'createReceptionResource').mockRejectedValueOnce(new Error(message))
+        await wrapper.vm.importLabware()
+        expect(mockShowAlert).toHaveBeenCalledWith(new Error(message), 'danger')
+      })
+
+      it('does not call scanIn LabWhere functionality', () => {
+        expect(scanBarcodesInLabwhereLocation).not.toHaveBeenCalled()
+      })
+    })
+  })
+
+  describe('Successful import', () => {
+    describe('Successfully import and scan in to LabWhere ', () => {
+      let mockedCreateReception
```

Comment:
This is more of a statement. You are pretty much mocking everything. Generally the only thing you need to mock are API calls as they are the only things you can't test directly. It would be worth double checking if everything needs to be mocked. It can cause false positives and things might not get tested correctly.